packages/plugins/experimental/plugin-map/src/hooks/useStabilizedObject.ts

```diff
@@ -0,0 +1,17 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { useState, useEffect } from 'react';
+
+export const useStabilizedObject = <T>(obj: T, customEqual: (oldObject: T, newObject: T) => boolean) => {
+  const [stableObj, setStableObj] = useState(obj);
+
+  useEffect(() => {
```

Comment:
beware that when using an effect here the returned markers above will be a render cycle behind the map and so if the map input changes there will be a render where the returned markers are for the previous map

Reply:
Ooof. Added a TODO comment for now. I'd also be down for trying to get a correct implementation of `useStabilizedObject`, but maybe not in this PR unless you know exactly what to do.